### ISSUE TITLE
Nyxt has window class "Sbcl"

### ISSUE THREAD

**Issue by franburstall** [atlas-engineer/nyxt#1227]
**Describe the bug**

According to `exwm`, nyxt windows have `WM_CLASS` set to "Sbcl".  They used to have class "nyxt".

**Precise recipe to reproduce the issue**
Fire nyxt under `exwm` and inspect the value of the emacs variable `exwm-class-name`.

**Information**
- OS name+version: Debian 10
- Graphics card and driver:
- Desktop environment / Window manager name+version: exwm
- How you installed Nyxt (Guix pack, package manager, build from source): built from source, master branch at f566fa89.

- Information from command copy-system-information:

Nyxt version: draft-do-not-use-me-11-gf566fa89 
Operating system kernel: Linux 4.19.0-14-amd64
Lisp implementation: SBCL 2.1.1
Features: (SLYNK WEBKIT2 WEBKIT2-2.30 WEBKIT2-TRACKING WEBKIT2-MUTE
           WEBKIT2-EMOJI WEBKIT2-MEDIA WEBKIT2-SANDBOXING GTK-3-22 GTK-3-20
           GTK-3-18 GTK-3-16 GTK-3-14 GTK-3-12 GTK-3-10 GTK-3-8 GTK-3-6 GTK-3-4
           GTK GDK-3-22 GDK-3-20 GDK-3-18 GDK-3-16 GDK-3-14 GDK-3-12 GDK-3-10
           GDK-3-8 GDK-3-6 GDK-3-4 CAIRO-1-10 CAIRO-1-12 GDK-PIXBUF GLIB-2-30
           GLIB-2-32 GLIB-2-34 GLIB-2-36 GLIB-2-38 GLIB-2-40 GLIB-2-42
           GLIB-2-44 GLIB-2-46 GLIB-2-48 GLIB-2-50 GLIB-2-52 GLIB-2-54
           GLIB-2-56 GLIB-2-58 GLIB FSET-EXT-STRINGS CUSTOM-HASH-TABLE-NATIVE
           SWANK PLUMP-UTF-32 GLOBAL-VARS DECLARE-TYPES PARENSCRIPT
           NAMED-READTABLES 21BIT-CHARS CHUNGA FLEXI-STREAMS CLOSER-MOP
           CL-PPCRE-UNICODE CL-UNICODE CL-PPCRE BORDEAUX-THREADS SPLIT-SEQUENCE
           GRAY-STREAMS FLAT-NAMESPACE X86-64 UNIX CFFI FLAT-NAMESPACE
           SEQUENCE-EMPTYP FAST-IO-SV FAST-IO SBCL-USES-SB-ROTATE-BYTE
           ASDF-SYSTEM-CONNECTIONS CL-JSON-CLOS CL-JSON THREAD-SUPPORT
           QUICKLISP ASDF3.3 ASDF3.2 ASDF3.1 ASDF3 ASDF2 ASDF OS-UNIX
           NON-BASE-CHARS-EXIST-P ASDF-UNICODE X86-64 GENCGC 64-BIT ANSI-CL
           COMMON-LISP ELF IEEE-FLOATING-POINT LINUX LITTLE-ENDIAN
           PACKAGE-LOCAL-NICKNAMES SB-LDB SB-PACKAGE-LOCKS SB-THREAD SB-UNICODE
           SBCL UNIX)

**Comment by Ambrevar** [comment]
I've noticed this too.

More precisely, the WM_CLASS is now `"nyxt", "Sbcl"` and for some reasy EXWM only picks the latter.

@aadcg @jeffbowman What do you think?  Any proper way to only set `nyxt`?  Maybe set `"nyxt", "Nyxt"`?

**Comment by Ambrevar** [comment]
See https://github.com/atlas-engineer/nyxt/pull/1174 for the relevant pull request that broke this.

**Comment by aadcg** [comment]
Some days(?) ago, a user mentioned the following on the nyxt IRC channel.

```
<mitigate> cragtegus cl-cffi-gtk got rid of g-set-prgname - you have to (setf
           g-prgname) instead 
```

**Comment by Ambrevar** [comment]
Nyxt uses Ferada's fork https://github.com/Ferada/cl-cffi-gtk/ so I
don't think it's the problem.

But thanks for mentioning because it could cause a build error, should
we update cl-cffi-gtk to Crategus' version.

**Comment by jeffbowman** [comment]
> I've noticed this too.
> 
> More precisely, the WM_CLASS is now `"nyxt", "Sbcl"` and for some reasy EXWM only picks the latter.
> 
> @aadcg @jeffbowman What do you think? Any proper way to only set `nyxt`? Maybe set `"nyxt", "Nyxt"`?

My first commit mentioned the WM_CLASS was set that way ("nyxt", "Sbcl"), I didn't see a way to change it at the time, but I can look again. If I recall correctly, the "Sbcl" value is pulled from the name of the executable or something, and wasn't something I could change. I'll revisit the docs and see if there is something I missed.

**Comment by franburstall** [comment]
>
> More precisely, the WM_CLASS is now "nyxt", "Sbcl" and for some reasy EXWM
> only picks the latter.


Looking more closely, I discover that WM_CLASS contains an instance name
and a class name.  In our case, the instance is "nyxt" and the class
"Sbcl".  You can access the instance with exwm-instance-name which is fine
for my purposes (which is to tell exwm that nyxt should run in char-mode).

**Comment by jeffbowman** [comment]
I may have found a solution, I'll submit a PR shortly

**Comment by Ambrevar** [comment]
I didn't know about exwm-instance-name, thanks for sharing!

**Comment by Ambrevar** [comment]
Fixed with #1229.
